src/compiler/utilities.ts

```diff
@@ -119,7 +119,12 @@ namespace ts {
     export function moduleResolutionIsEqualTo(oldResolution: ResolvedModuleFull, newResolution: ResolvedModuleFull): boolean {
         return oldResolution.isExternalLibraryImport === newResolution.isExternalLibraryImport &&
             oldResolution.extension === newResolution.extension &&
-            oldResolution.resolvedFileName === newResolution.resolvedFileName;
+            oldResolution.resolvedFileName === newResolution.resolvedFileName &&
+            (oldResolution.packageId === newResolution.packageId || oldResolution.packageId && newResolution.packageId && packageIdIsEqual(oldResolution.packageId, newResolution.packageId));
+    }
+
+    function packageIdIsEqual(a: PackageId, b: PackageId): boolean {

```

Comment:
Possibly this is following a precedent, but why not have `packageIdIsEqual` handle `undefined`?